src/types.jl

```diff
@@ -899,3 +950,79 @@ end
 
 # Take precedence over AtomsBase.jl show function
 Base.show(io::IO, ::MIME"text/plain", s::Union{System, ReplicaSystem}) = show(io, s)
+
+
+
+# Convert AtomsBase AbstractSystem to Molly system
+"""
+Converts AtomsBase `AbstractSystem` to Molly `System``. To add kwargs not present in the
+AtomsBase interface (e.g. pair potentials) call the convenience constructor:
+
+`System(sys::System)`
+"""
+function System(sys::AbstractSystem{D}) where D
+
+    #Convert BC to Molly types
+    bb = bounding_box(sys)
+    boundary_conditions = AtomsBase.boundary_conditions(sys)
+    
+    #Check if box is cubic
+    angles = []
+    for i in range(1,D)
+        for j in range(i+1,D)
+            push!(angles, ustrip(dot(bb[i],bb[j])))
+        end
+    end
+    isCubic = all(angles .== 0.0)
+
+    if !isinfinite(sys)
+        box_lengths = norm.(bb)
+    end
+    box_lengths = convert(Vector, box_lengths) #was SVector need to be mutable
+
+    if any(typeof.(boundary_conditions) .== DirichletZero)
+        raise(ArgumentError("Molly does not support DirichletZero boundary conditions."))
+    end
+
+    if isCubic && D == 2
+        @warn "Molly RectangularBoundary assumes origin @ (0, 0, 0)"
+        molly_boundary = RectangularBoundary(box_lengths...)
+    elseif isCubic && D == 3
+        @warn "Molly CubicSystem assumes origin @ (0, 0, 0)"
+        molly_boundary = CubicBoundary(box_lengths...)
+    elseif D == 3
+        @warn "Molly TriclinicBoundary assumes origin @ (0, 0, 0)"
+        if any(ustrip.(box_lengths) .== Inf)
+            throw(ArgumentError("Triclinic domain does not support infinite boundaries"))
+        end
+        molly_boundary = TriclinicBoundary(bb...)
+    else
+        throw(ArgumentError("Molly does not support 2D triclinic domains"))
+    end
+
+    idx = 0
+    atoms = map(sys) do atom
```

Comment:
`for` loop seems clearer if you are incrementing an index.

Reply:
Have to define an empty array the other way and then push to it. This is less lines of code. Happy to change it though.

Reply:
It may not be safe in future Julia versions, which could parallelise `map` or otherwise change the execution order.